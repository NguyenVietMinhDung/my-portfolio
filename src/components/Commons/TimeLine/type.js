import type { Props as Item } from './components/Event/type';

export type Props = {
  iconClassName: string,
  items: Array<Item>,
};
