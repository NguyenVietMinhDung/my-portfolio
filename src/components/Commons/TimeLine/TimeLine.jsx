// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './TimeLine.scss';
import { Props } from './type';
import Event from './components/Event';

const TimeLine = (props: Props) => {
  const { iconClassName, items } = props;
  return (
    <div styleName="timeline">
      <div styleName="icon">
        <div styleName="contour">
          <i className={`${iconClassName} size-sm`} />
        </div>
      </div>
      <ul styleName="events">
        {items.map(item => <Event {...item} />)}
      </ul>
    </div>
  );
};

export default memo<Props>(CSSModules(TimeLine, styles));
