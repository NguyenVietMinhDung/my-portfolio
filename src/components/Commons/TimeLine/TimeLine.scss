@import '../../../styles/abstracts/variables';

.timeline {
	position: relative;

	display: flex;
	flex-flow: column wrap;
	justify-content: center;

	.icon {
		display: flex;
		flex-flow: row wrap;

		.contour {
			display: flex;

			width: 50px;
			height: 50px;

			border: 4px solid $grey-color;
			border-radius: 100%;
			align-items: center;
			flex-flow: row wrap;
			justify-content: center;

			i {
				color: $violet-color;
			}
		}
	}

	&::before {
		position: absolute;
		bottom: 0;
		left: 27px;

		width: 4px;
		height: calc(100% - 58px);

		content: '';

		background: $grey-color linear-gradient(to bottom, $grey-color 1%, $grey-color 75%, $opalescent-color 100%);
	}

	.events {
		display: flex;

		padding-left: 58px;
		flex-flow: column wrap;
	}
}
