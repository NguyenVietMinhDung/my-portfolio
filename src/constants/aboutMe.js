export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
