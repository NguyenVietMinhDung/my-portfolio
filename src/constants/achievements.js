export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
