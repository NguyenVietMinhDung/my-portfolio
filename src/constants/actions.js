export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
