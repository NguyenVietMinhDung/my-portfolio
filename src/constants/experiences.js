export const TITLE = 'Experiences';
export const timeline = {
  iconClassName: 'lni-briefcase',
  items: [
    {
      key: 1,
      organization: 'Ubitec',
      role: 'JAVASCRIPT DEVELOPER',
      startDate: 'September 2019',
      endDate: 'May 2020',
    },
    {
      key: 2,
      organization: 'KMS Technology Inc., Vietnam',
      role: 'SOFTWARE ENGINEER',
      startDate: 'November 2017',
      endDate: 'August 2019',
    },
    {
      key: 3,
      organization: 'KMS Technology Inc., Vietnam',
      role: 'FRESHER',
      startDate: 'September 2017',
      endDate: 'November 2017',
      description: 'I had joined KMS as a fresher before graduating. '
      + 'I had been trained to become an official employee for 2 months. '
      + 'Then, I joined one of the biggest project team in the company.',
    },
  ],
};
