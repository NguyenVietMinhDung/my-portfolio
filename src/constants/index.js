import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
