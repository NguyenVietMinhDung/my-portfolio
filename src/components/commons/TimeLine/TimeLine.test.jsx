import React from 'react';
import { shallow } from 'enzyme';
import { timeline } from '../../../constants/experiences';
import TimeLine from './TimeLine';

const { items } = timeline;

const defaultProps = {
  items,
};

describe('<TimeLine />', () => {
  test('renders correctly', () => {
    const component = shallow(<TimeLine {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
