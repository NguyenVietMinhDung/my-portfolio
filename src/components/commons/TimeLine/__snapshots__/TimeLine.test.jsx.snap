// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeLine /> renders correctly 1`] = `
<div
  className="timeline"
>
  <div
    className="icon"
  >
    <div
      className="contour"
    >
      <i
        className="undefined size-sm"
      />
    </div>
  </div>
  <ul
    className="events"
  >
    <Memo(Event)
      endDate="May 2020"
      key="1"
      organization="Ubitec"
      role="JAVASCRIPT DEVELOPER"
      startDate="September 2019"
    />
    <Memo(Event)
      endDate="August 2019"
      key="2"
      organization="KMS Technology Inc., Vietnam"
      role="SOFTWARE ENGINEER"
      startDate="November 2017"
    />
    <Memo(Event)
      description="I had joined KMS as a fresher before graduating. I had been trained to become an official employee for 2 months. Then, I joined one of the biggest project team in the company."
      endDate="November 2017"
      key="3"
      organization="KMS Technology Inc., Vietnam"
      role="FRESHER"
      startDate="September 2017"
    />
  </ul>
</div>
`;
